src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectISizeDistributionSpark.java

```diff
@@ -0,0 +1,175 @@
+    public boolean nonSecondaryAlignmentsOnly = true;
+
+    @Argument(shortName = "SS",
+            fullName = "nonSupplementaryAlignmentsOnly",
+            doc = "If set to true, filter out supplementary alignments reads. Default value: true.")
+    public boolean nonSupplementaryAlignmentsOnly = true;
+
+    @Argument(shortName = "F",
+            fullName = "whichEndOfPairToUse",
+            doc = "Which end of pairs to use for collecting information. Possible values:{0,1,2}, where 1 and 2 " +
+                    "stands for first or second end respectively. Option 0 picks up information from 1st end " +
+                    "when both ends are available, and pick either one when only one end is available. " +
+                    "(Remember, in SV analysis, a truncated region may be investigated, so this is possible.)" +
+                    " Unused if only paired reads are used (properlyPairedOnly==true).")
+    public int whichEndOfPairToUse = 1;
+
+    @Argument(shortName = "Q",
+            fullName = "MQPassingThreshold",
+            doc = "If set non-zero value, only include reads passing certain mapping quality threshold.")
+    public double MQPassingThreshold = 0.0;
+
+    @Argument(shortName = "Qm",
+            fullName = "mateMQPassingThreshold",
+            doc = "If set to non-zero value, only include reads with mate passing certain mapping quality threshold.")
+    public double mateMQPassingThreshold = 0.0; // TODO: efficient way to test
+
+
+    // path to Picard R script for producing histograms in PDF files.
+    private static final String R_SCRIPT = "insertSizeHistogram.R";
+
+    // should this be static?
+    // Calculates InsertSizeMetrics for all METRIC_ACCUMULATION_LEVELs provided
+    private static ISZMetricsCollectorSpark collector;
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<GATKRead> reads = getReads();
+
+        collector = new ISZMetricsCollectorSpark(reads, METRIC_ACCUMULATION_LEVEL, getHeaderForReads().getReadGroups(), DEVIATIONS_TOL);
+
+        if(PRODUCE_FILES)
+            saveResults(getHeaderForReads(), getReadSourceName());
+    }
+
+    @Override
+    public ReadFilter makeReadFilter() { // return type serializable. implicitly called in getReads(): behavior from GATKTools
+        final boolean filters[] = {properlyPairedOnly, mappedReadsOnly, mateMappedReadsOnly,
+                                   nonDupReadsOnly, nonSecondaryAlignmentsOnly, nonSupplementaryAlignmentsOnly};
+
+        final SVCustomReadFilter svfilter = new SVCustomReadFilter(filters, whichEndOfPairToUse,
+                                                                   MQPassingThreshold, mateMQPassingThreshold);
+
+        final AlignmentAgreesWithHeaderReadFilter alignmentAgreesWithHeader = new AlignmentAgreesWithHeaderReadFilter(getHeaderForReads());
+
+        return svfilter.and(alignmentAgreesWithHeader); // Is this bad practice (type mixture)?
```

Comment:
this is fine, but do document that you are explicitly *not* calling super because you don't want to use the `WellformedReadFilter` (assuming not `WellformedReadFilter` is in fact the right thing here, is it?)